Backend/src/main/java/com/harmonia/backend/controller/UserController.java

```diff
@@ -30,24 +50,48 @@ public Iterable<UserResponse> listUsers(@RequestParam(name = "username", require
         return userService.listUsers();
     }
 
+    /**
+     * Get a list of users by username
+     *
+     * @param userName the username to search for
+     * @return an iterable list of UserResponse objects
+     */
     @GetMapping("username/{username}")
     @CrossOrigin
     public Iterable<UserResponse> listUsersByUserName(@PathVariable("username") String userName) {
         return userService.listUsersByUserName(userName);
     }
 
-    @GetMapping("/{userId}")
+    /**
+     * Get a user by their user ID
+     *
+     * @param userId the ID of the user to retrieve
+     * @return a UserResponse object
+     */
+    @GetMapping("userId/{userId}")
     @CrossOrigin
     public UserResponse findUserById(@PathVariable Long userId) {
         return userService.findUserById(userId);
     }
 
+    /**
+     * Create a new user
+     *
+     * @param user the CreateUserRequest object containing the user data
+     * @return a UserResponse object
+     */
     @PostMapping
     public UserResponse createUser(@RequestBody CreateUserRequest user) {
         return userService.createUser(user);
     }
 
 
+    /**
+     * Authenticate a user by their username and password
+     *
+     * @param credentials a Map containing the username and password of the user
+     * @return a ResponseEntity with a UserResponse object if the user is authenticated, or a not found response if not
+     */
     @PostMapping("login")
     public ResponseEntity<UserResponse> loginUser(@RequestBody Map<String, String> credentials) {
```

Comment:
Maybe replace the (@RequestBody Map<String, String> credentials) for a DTO  UserCredentialsPO